### PR TITLE
Add proactive inactivity checks to the Generic Assistant

### PR DESCRIPTION
**Pull request by Yashi11**
Adds persona-aware, LLM-generated activity checks before an inactive voice session is disconnected.

  **Changes**:

-   Added a reusable activity-check processor that starts a timer after bot speech finishes and resets on any user speech.
-   Added a 10-minute first inactivity threshold and a 30-second final check threshold, configurable through environment variables or session settings.
-   Uses LLM and TTS for both proactive utterances, then disconnects only after the closing audio completes.
-   Added activity-check lifecycle logs for arming, firing, cancellation, and disconnect.
-   Added Compose environment pass-through and documented configuration defaults.

  **Testing**

-   uv run --project . --group dev ruff format --check src/examples/generic/pipeline.py src/examples/shared/activity_check.py
-   uv run --project . --group dev ruff check src/examples/generic/pipeline.py src/examples/shared/activity_check.py
-   python -m compileall -q src/examples/generic/pipeline.py src/examples/shared/activity_check.py
-   Manual verification through the Compose-deployed Generic Assistant: greeting, timer arm, first check, reset on user speech, second closing check, and disconnect.

## Summary by CodeRabbit

- **New Features**
  - Added proactive inactivity monitoring with a two-stage warning flow for the generic assistant experience.
  - Automatically enables inactivity checks in the generic assistant pipeline when configured.
  - Added per-example timing configuration for inactivity checks.

- **Improvements**
  - Standardized the default pipeline TLS setting across services.

- **Tests**
  - Added coverage for warning progression, timing edge cases, cancellation, context behavior, and configuration validation.

### CODE REVIEW

**Comment by Yashi11** [comment]
https://github.com/user-attachments/assets/d54253d0-294e-4338-a71a-4145c01ed336

Can extend this to other examples as well - for now I have restricted it to generic only.

File: src/examples/generic/pipeline.py [hunk 1]
```diff
@@ -164,6 +165,56 @@ async def bot(runner_args: RunnerArguments) -> None:
 
     audio_recorder = create_audio_recorder()
 
+    activity_check_enabled = str(
+        body.get("activity_check_enabled", os.getenv("ACTIVITY_CHECK_ENABLED", "true"))
```
Comment (AshutoshRautela) [inline src/examples/generic/pipeline.py]: ACTIVITY_CHECK_ENABLED
if enabled, this will be enabled for all examples in future ?

File: src/examples/generic/pipeline.py [hunk 2]
```diff
@@ -164,6 +165,56 @@ async def bot(runner_args: RunnerArguments) -> None:
...
+    ).lower() not in {"0", "false", "no", "off"}
+
+    def _timeout(body_key: str, env_key: str, default: float) -> float:
```
Comment (AshutoshRautela) [inline src/examples/generic/pipeline.py]: I think this part can be extracted to a separate file @Yashi11

File: src/examples/generic/pipeline.py [hunk 3]
```diff
@@ -164,6 +165,56 @@ async def bot(runner_args: RunnerArguments) -> None:
...
+        return value
+
+    activity_check_interval_s = _timeout("activity_check_interval_s", "ACTIVITY_CHECK_INTERVAL_S", 600.0)
+    first_warning_s = _timeout("first_warning_s", "FIRST_WARNING_S", activity_check_interval_s)
+    second_warning_s = _timeout("second_warning_s", "SECOND_WARNING_S", 30.0)
+    warning_completion_timeout_s = _timeout(
+        "warning_completion_timeout_s", "ACTIVITY_WARNING_COMPLETION_TIMEOUT_S", 45.0
+    )
+
+    async def on_activity_warning(stage: int) -> None:
```
Comment (AshutoshRautela) [inline src/examples/generic/pipeline.py]: Same for this one. Can be extracted to a separate file to keep the pipeline.py clean

Comment (AshutoshRautela) [reply src/examples/generic/pipeline.py]: Maybe can be moved out to activity_check.py

File: .env.example [hunk 4]
```diff
@@ -20,6 +20,11 @@ NVIDIA_API_KEY=nvapi-...
 # Optional Pipecat pipeline idle timeout in seconds.
 # Minimum is 300 seconds.
 # PIPELINE_IDLE_TIMEOUT_SECS=600
+# ACTIVITY_CHECK_ENABLED=true
```
Comment (AshutoshRautela) [inline .env.example]: Would not prefer to add too many vars to env.example. Would prefer to keep .env config short and simple

File: src/examples/generic/pipeline.py (hunk 1 above)
Comment (Yashi11) [reply src/examples/generic/pipeline.py]: I have only applied it to generic example for now - but we can extend it to the others as well based on the feedback

File: .env.example (hunk 4 above)
Comment (Yashi11) [reply .env.example]: We can have this as the default behavior of the pipeline - so we will move from .env to directly bot.py - what do you think?

Comment (dharmendrach) [reply .env.example]: We might need to create example_config.yaml for all such params, or we can use top-level example_registry.yaml. @AshutoshRautela, what do you think?

File: tests/test_activity_check.py [hunk 5]
```diff
@@ -0,0 +1,81 @@
+# SPDX-FileCopyrightText: Copyright (c) 2024-2026, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: BSD-2-Clause
+
```
Comment (dharmendrach) [inline tests/test_activity_check.py]: This should be under unit folder

File: src/examples/generic/pipeline.py [hunk 6]
```diff
@@ -242,7 +294,14 @@ async def on_breakdown(observer, breakdown):
             enable_metrics=True,
             enable_usage_metrics=True,
         ),
-        idle_timeout_secs=runner_args.pipeline_idle_timeout_secs,
+        idle_timeout_secs=(
```
Comment (dharmendrach) [inline src/examples/generic/pipeline.py]: Idle timeout can be the default; if required, use a higher timeout. The proactivity processor should trigger at 30- and 60 secs interval. We can generate an explicit event for stopping the pipeline from the Proactivity processor.

**Review by dharmendrach — CHANGES_REQUESTED** [review]

File: .env.example (hunk 4 above)
Comment (AshutoshRautela) [reply .env.example]: image
image
I would say keep example_registry.yaml. We do have example specific config today, but we have prepared them as generic and can be carried it to any examples later on. For now @Yashi11 you can add it to examples_registry.yaml. 
Later if config requirements become too big, we'll consider a separate file for it

File: src/examples/generic/pipeline.py (hunk 1 above)
Comment (AshutoshRautela) [reply src/examples/generic/pipeline.py]: @Yashi11  Take a look at examples_registry.yaml. It'll help to isolate it per example

**Review by AshutoshRautela — COMMENTED** [review]
Minor improvements required regarding restructuring. Rest looks good

**Review by dharmendrach — DISMISSED** [review]

**Review by dharmendrach — APPROVED** [review]
